webui/react/src/components/TaskActionDropdown.tsx

```diff
@@ -53,6 +53,9 @@ const TaskActionDropdown: React.FC<Props> = ({ task }: Props) => {
       switch (params.key) { // Cases should match menu items.
         case 'kill':
           await killTask(task);
+          if (task.type === TaskType.Experiment) {
+            await updateExperimentLocally(exp => ({ ...exp, state: RunState.StoppingCanceled }));
+          } // we dont provide the immediate update for command types yet.
```

Comment:
nit thoughts: How about conforming comments to full sentences, so starting with a capitalized letter and ending with a punctuation (which you have)? The explanation comment is great, it gives us a reason why code is "missing".  How about placing the comment right after the `if` block on a separate line so it's not on the same line as the control structure?
micro nit: `don't`